inference-explainers/inference_api.md

```diff
@@ -0,0 +1,252 @@
+- **tensor_shape:** An array specifying the shape of the tensor.
+- **tensor_content:** A flattened row-major representation of the tensor values.
+- **tensor_name:** Specifies the tensor's name to match the model's signature. This value is ignored for PyTorch models.
+
+**tensor_content** values are represented as strings, with plans to eventually enable storage as numeric types, as numeric value support is currently under development.
+
+The following illustrates a batch request involving two models, “pcvr_v1” and “pcvr_v2”:
+
+```json
+{
+  "request": [
+    {
+      "model_path": "pcvr_v1",
+      "tensors": [
+        {
+          "tensor_name": "feature1",
+          "data_type": "DOUBLE",
+          "tensor_shape": [2, 1],
+          "tensor_content": ["0.454920", "-0.25752"]
+        }
+      ]
+    },
+    {
+      "model_path": "pcvr_v2",
+      "tensors": [
+        {
+          "tensor_name": "feature1",
+          "data_type": "INT32",
+          "tensor_shape": [2, 1],
+          "tensor_content": ["5", "6"]
+        },
+        {
+          "tensor_name": "feature2",
+          "data_type": "FLOAT",
+          "tensor_shape": [2, 2],
+          "tensor_content": ["0.5", "0.6", "0.7", "0.8"]
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### Batch response
+
+The batch inference request is structured as a JSON object containing a top-level array. Each element within this array represents an individual response represented as a JSON object. This array consists of at least one element. If a batch inference fails, this one element represents the batch error.
+
+Each individual response object signifies either successful inference output or an error that was encountered during inference.
+
+- For a success inference, each tensor JSON object follows the same format as in the request case except that **tensor_content** directly contains numericals instead of strings.
+
+- In the failure case, the **model_path** field is optional. It is populated when the inference directed towards a specific model fails and is left empty when an entire batch request fails. The response has a mandatory **error** field with each error composed of a JSON object with **error_type** and **description** string fields.
+  We currently report the following error types:
+
+  - UNKNOWN
+  - INPUT_PARSING
+  - MODEL_NOT_FOUND
+  - MODEL_EXECUTION
+  - OUTPUT_PARSING
+  - GRPC
+
+The **description** field propagates back the original Abseil error message from the inference sidecar C++ program.
+
+The following are some illustrative scenarios for reference:
+
+- **Batch with all successful outputs:** The top-level array contains multiple response objects, each with the **model_path** and **tensor** fields populated which indicates successful inference for each input in the batch:
+
+```json
+{
+  "response": [
+    {
+      "model_path": "pcvr_1",
+      "tensors": [
+        {
+          "tensor_name": "StatefulPartitionedCall:0",
```

Comment:
Nit: Let's use "PartitionedCall:0" assuming that the model is frozen.

Reply:
Done